runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -356,4 +358,76 @@ J9::ARM64::TreeEvaluator::BNDCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    cg->decReferenceCount(secondChild);
    secondChild->setIsNonNegative(true);
    return (NULL);
+}
+
+static void VMoutlinedHelperArrayStoreCHKEvaluator(TR::Node *node, TR::Register *srcReg, TR::Register *dstReg, TR::CodeGenerator *cg)
+   {
+   TR::SymbolReference *arrayStoreChkHelper = node->getSymbolReference();
+
+   TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(2, 2, cg->trMemory());
+   TR::addDependency(deps, dstReg, TR::RealRegister::x0, TR_GPR, cg);
+   TR::addDependency(deps, srcReg, TR::RealRegister::x1, TR_GPR, cg);
+
+   TR::Instruction *gcPoint = generateImmSymInstruction(cg, TR::InstOpCode::bl, node,
+                                                        (uintptr_t)arrayStoreChkHelper->getMethodAddress(),
+                                                        deps, arrayStoreChkHelper, NULL);
+   gcPoint->ARM64NeedsGCMap(cg, 0xFFFFFFFF);
+
+   cg->machine()->setLinkRegisterKilled(true);
+   }
+
+static void VMoutlinedHelperWrtbarEvaluator(TR::Node *node, TR::Register *srcReg, TR::Register *dstReg, TR::CodeGenerator *cg)
+   {
+   auto gcMode = TR::Compiler->om.writeBarrierType();
+   if (gcMode == gc_modron_wrtbar_none)
+      return;
+
+   TR::Compilation *comp = TR::comp();
+   TR::LabelSymbol *doneLabel = generateLabelSymbol(cg);
+   TR::SymbolReference *wbref = comp->getSymRefTab()->findOrCreateWriteBarrierStoreSymbolRef(comp->getMethodSymbol());
+
+   TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(2, 2, cg->trMemory());
+   TR::addDependency(deps, dstReg, TR::RealRegister::x0, TR_GPR, cg);
+   TR::addDependency(deps, srcReg, TR::RealRegister::x1, TR_GPR, cg);
+
+   generateCompareBranchInstruction(cg, TR::InstOpCode::cbzx, node, srcReg, doneLabel);
+
+   TR::Instruction *gcPoint = generateImmSymInstruction(cg, TR::InstOpCode::bl, node,
+                                                        (uintptr_t)wbref->getMethodAddress(),
+                                                        new (cg->trHeapMemory()) TR::RegisterDependencyConditions((uint8_t)0, 0, cg->trMemory()),
+                                                        wbref, NULL);
+   gcPoint->ARM64NeedsGCMap(cg, 0xFFFFFFFF);
+
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, doneLabel, deps);
+
+   cg->machine()->setLinkRegisterKilled(true);
+   }
+
+TR::Register *
+J9::ARM64::TreeEvaluator::ArrayStoreCHKEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
```

Comment:
To make this easier to read, can you store `node->getFirstChild()` into a `wrtBarNode` and use that instead?